packages/cardpay-sdk/sdk/utils/general-utils.ts

```diff
@@ -381,3 +387,26 @@ export async function wait(ms = 1000) {
     setTimeout(resolve, ms);
   });
 }
+
+// Returns the current standard gas price in wei in the specified gas token
+export async function currentGasPrice(network: string, gasTokenAddress: string) {
+  let gasStationResponse = await fetch(`${getConstantByNetwork('relayServiceURL', network)}/v1/gas-station/`, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    },
+  });
+
+  let gasPriceInNativeToken = (await gasStationResponse.json()).standard;
+
+  let exchangeAddress = getAddressByNetwork('scheduledPaymentExchange', network);
+  let wrappedNativeTokenAddress = getAddressByNetwork('wrappedNativeToken', network);
+  const provider = new JsonRpcProvider(getConstantByNetwork('rpcUrl', network));
+  let contract = new EthersContract(exchangeAddress, ExchangeABI, provider);
+
+  let gasTokenInUsd = (await contract.exchangeRateOf(gasTokenAddress)).value;
+  let wrappedNativeTokenInUsd = (await contract.exchangeRateOf(wrappedNativeTokenAddress)).value;
+
+  return wrappedNativeTokenInUsd.div(gasTokenInUsd).mul(gasPriceInNativeToken);
```

Comment:
keep in mind these are _really_ large numbers we are dealing with, because they are in `wei`. in the past I have had issues with order of operations concerning dividing very large numbers that result in loss of precision. It might be better to first perform your multiplication before the divide so that you are dividing a very large number by another very large number instead of dividing a normal size number by a very large number.